morph/react/block-list.js

```diff
@@ -16,10 +16,38 @@ export function enter(node, parent, state) {
   state.render.push(
     `{Array.isArray(${value}) && ${value}.map((item, index, list) => `
   )
+
+  if (parent.viewPath) {
+    let itemContextName =
+      getProp(node, 'itemContextName') || defaultItemContextName(state)
+    state.render.push(
+      `<fromData.DataProvider
+        context="${itemContextName.value}"
+        value={{ ${itemContextName.value}: item }}
```

Comment:
let's wrap this in quotes `"${itemContextName.value}": item` to ensure it morphs alright weird chars 🙈